### PR TITLE
Update HangSeng.com

### PR DESCRIPTION
**Pull request by Marco129**
Pages at http://www.hangseng.com/cms/ are HTTP only.
HTTPS version will return 404 Not Found.

### CODE REVIEW

File: src/chrome/content/rules/HangSengBank.xml [hunk 1]
```diff
@@ -14,17 +14,18 @@
...
 	Exceptions: 
-	- HSBCnet logon page
+	- CMS (404 Not Found)
 	-->
-	<exclusion pattern="^http://www\.hangseng\.com\/cms\/sidnet\/files\/hsbcnet" />
+	<exclusion pattern="^http://www\.hangseng\.com\/cms\/" />
 
 	<rule from="^http://hangseng\.com/"
-		to="https://www.hangseng.com/" />
+		to="https://bank.hangseng.com/" />
```
Comment (J0WI) [inline src/chrome/content/rules/HangSengBank.xml]: The redirect from www to bank works fine on the servers side, so we should not touch it.

**Comment by jeremyn** [comment]
@Marco129 Are you still interested in working on this one?

**Comment by Marco129** [comment]
Yes, PR updated.

**Comment by jeremyn** [comment]
Great! Please make the following changes:

Mark as `Invalid certificate`
- [ ] https://promotion.hangseng.com
- [ ] https://hangseng.com.cn
- [ ] https://www.ecds.hangseng.com.cn
  
Mark as `No working URL known`:
- [ ] https://fileupload.hangseng.com
- [ ] https://fileupload1.hangseng.com
- [ ] https://fileupload2.hangseng.com
- [ ] https://www.ipoonline.hangseng.com
 
Mark as `Refused`:
- [ ] https://funds.hangseng.com.cn
 
Mark as `Time out`:
- [ ] https://eba.hangseng.com
- [ ] https://evalue.hangseng.com
- [ ] https://inv.hangseng.com
- [ ] https://realestate.hangseng.com
- [ ] https://sc.hangseng.com
- [ ] https://sw.hangseng.com
 
Add `target`:
- [ ] https://www.hangseng.com (see under "Also" below)
- [ ] https://quote.hangseng.com
- [ ] https://www.hangseng.com.cn
 
Also:
- [ ] Add a `test` for https://quote.hangseng.com/streaming/landing directly under the `quote.` `target`, with one extra indent
- [ ] URLs under https://www.hangseng.com/cms/ seem okay now and I don't get an `incomplete certificate chain`. Can you please confirm and then remove `www.hangseng.com` from the comment and delete the `Contents at HTTP` category?

**Comment by jeremyn** [comment]
The `url` in the `test` should be `http`:
```xml
		<test url="http://quote.hangseng.com/streaming/landing/" />
```
That should fix the failing Travis check.

Also, please switch the order of `promotion.hangseng.com` and `www.cn.hangseng.com` in the comment. For reference, our preferred sort order is for domains to be in alphabetical order starting from the top level domain at the right reading left, moving `^` and `www` to the top of their group. For example:
```
example.com
www.example.com
a.example.com
www.a.example.com
b.a.example.com
b.example.com
example.net
www.example.net
a.example.net
```
In this case, `cn` should come before `promotion`.

**Comment by jeremyn** [comment]
Thanks, merged.

**Pull request merged**